File: src/main/webapp/src/components/FactoryService.js
angular.module('soutenanceplanner.factory', []);

angular.module('soutenanceplanner.factory')

.service('FactoryService', [ '$http', 'WS_SERVER_URL',

	function($http, WS_SERVER_URL) {

		var FactoryService = {

			user: function(){
				return $http({
					method: 'GET',
					url: WS_SERVER_URL + "/factory/user",
					data: {}
				});
			},

			formation: function(){
				return $http({
					method: 'GET',
					url: WS_SERVER_URL + "/factory/formation",
					data: {}
				});
			},

			calendar: function(){
				return $http({
					method: 'GET',
					url: WS_SERVER_URL + "/factory/calendar",
					data: {}
				});
			},

			authenticate: function(){
				return $http({
					method: 'GET',
					url: WS_SERVER_URL + "/factory/authenticate",
					data: {}
				});
			}

		};

		return FactoryService;
	}
]);

File: src/main/java/com/angers/m2sili/soutenance/service/FactoryService.java
package com.angers.m2sili.soutenance.service;

import com.angers.m2sili.soutenance.model.Calendar;
import com.angers.m2sili.soutenance.model.Formation;
import com.angers.m2sili.soutenance.web.dto.AuthenticateDTO;
import com.angers.m2sili.soutenance.web.dto.UserDTO;
import com.angers.m2sili.soutenance.model.Oral;

/**
 * Interface du service de Factory des beans.
 * 
 * @author typhoon
 *
 */

public interface FactoryService {

	public UserDTO user();

	public Formation formation();
	
	public AuthenticateDTO authenticate();

	public Oral oral();

	public Calendar calendar();

}
